pkg/river/encoding/block.go

```diff
@@ -37,38 +37,20 @@ func (bf *blockField) convertBlock(namePrefix []string, reflectValue reflect.Val
 		}
 		reflectValue = reflectValue.Elem()
 	}
+	if reflectValue.Kind() != reflect.Struct {
```

Comment:
What happened in this file? Looks like a bad merge; we don't want to revert the code here.